src/Umbraco.Web/Compose/NestedContentPropertyComponent.cs

```diff
@@ -0,0 +1,180 @@
+            {
+                // When a content node contains nested content property
+                // Check if the copied node contains a nested content
+                var nestedContentProps = entity.Properties.Where(x => x.PropertyType.PropertyEditorAlias == Constants.PropertyEditors.Aliases.NestedContent);
+
+                // Each NC Property on a doctype
+                foreach (var nestedContentProp in nestedContentProps)
+                {
+                    // A NC Prop may have one or more values due to cultures
+                    var propVals = nestedContentProp.Values;
+                    foreach (var cultureVal in propVals)
+                    {
+                        // Remove keys from published value & any nested NC's
+                        var updatedPublishedVal = CreateMissingNestedContentKeys(cultureVal.PublishedValue?.ToString());
+                        cultureVal.PublishedValue = updatedPublishedVal;
+
+                        // Remove keys from edited/draft value & any nested NC's
+                        var updatedEditedVal = CreateMissingNestedContentKeys(cultureVal.EditedValue?.ToString());
+                        cultureVal.EditedValue = updatedEditedVal;
+                    }
+                }
+            }
+        }
+
+        public void Terminate()
+        {
+            ContentService.Copying -= ContentService_Copying;
+            ContentService.Saving -= ContentService_Saving;
+            ContentService.Publishing -= ContentService_Publishing;
+        }
+
+        private string CreateNewNestedContentKeys(string ncJson)
+        {
+            if (string.IsNullOrWhiteSpace(ncJson))
+                return ncJson;
+
+            // Try & convert JSON to JArray (two props we will know should exist are key & ncContentTypeAlias)
+            var ncItems = JArray.Parse(ncJson);
+
+            // NC prop contains one or more items/rows of things
+            foreach (var nestedContentItem in ncItems.Children<JObject>())
+            {
+                foreach (var ncItemProp in nestedContentItem.Properties())
+                {
+                    if(ncItemProp.Name.ToLowerInvariant() == "key")
+                        ncItemProp.Value = Guid.NewGuid().ToString();
+
+                    // No need to check this property for JSON - as this is a JSON prop we know
+                    // That onyl contains the string of the doctype alias used as the NC item
+                    if (ncItemProp.Name == NestedContentPropertyEditor.ContentTypeAliasPropertyKey)
+                        continue;
+
+                    // As we don't know what properties in the JSON may contain the nested NC
+                    // We are detecting if its value stores JSON to help filter the list AND that in its JSON it has ncContentTypeAlias prop
+                    if (ncItemProp.Value.ToString().DetectIsJson() && ncItemProp.Value.ToString().Contains(NestedContentPropertyEditor.ContentTypeAliasPropertyKey))
```

Comment:
This isn't a very nice way of recursing. I realize this will 'work' but what if someone types in the value of "ncContentTypeAlias" in one of their textbox values? This will think it needs to recurse and will probably explode because we aren't error handling the serialized value when trying to parse into JSON. Also this will not work if someone has a nested content cell with another non-nested content editor that also contains a nested content editor. 

IMO the way to do this is to 'just' recurse with the `ncItemProp.Value.ToString()` (if it's not null), don't do this check since it will be unnecessary and properly parse out the json defensively. But you would need to keep recursing even if it's not nested content and you'd probably want to recurse on a JToken not a string. 

There's probably a nicer way to do that too and use JsonPath instead https://www.newtonsoft.com/json/help/html/QueryJsonSelectTokenJsonPath.htm to select all tokens that you are looking for and then update from there. If you can make that work that would be far better.